### PR TITLE
Support encrypting keys

### PR DESCRIPTION
**Pull request by lmars**
First part of #5 (the next part will be support for changing passphrases and enabling / disabling encryption)

### CODE REVIEW

File: encrypted/encrypted.go [hunk 1]
```diff
@@ -174,6 +174,15 @@ func Encrypt(plaintext, passphrase []byte) ([]byte, error) {
 	return json.Marshal(data)
 }
 
+// Marshal encrypts the JSON encoding of v using passphrase.
+func Marshal(v interface{}, passphrase []byte) ([]byte, error) {
+	data, err := json.Marshal(v)
```
Comment (titanous) [inline encrypted/encrypted.go]: Could `MarshalIndent` here.

File: local_store.go [hunk 2]
```diff
@@ -230,45 +244,107 @@ func (f *fileSystemStore) Commit(meta map[string]json.RawMessage, consistentSnap
...
+			return err
+		}
+		pk.Encrypted = true
+	} else {
+		pk.Data, err = json.MarshalIndent(keys, "", "  ")
+		if err != nil {
+			return err
+		}
+	}
+	data, err := json.MarshalIndent(pk, "", "  ")
```
Comment (titanous) [inline local_store.go]: Use `\t` for `MarshalIndent`

**Comment by titanous** [comment]
Neat! We should also add support for environment variables like `TUF_ROOT_PASSPHRASE`, `TUF_SNAPSHOT_PASSPHRASE`, etc.

Also, don't forget to add a flag to disable encryption when generating keys, either in this patch or the next.

**Comment by lmars** [comment]
@titanous comments addressed

File: tuf/main.go [hunk 3]
```diff
@@ -1,24 +1,31 @@
...
 func main() {
 	log.SetFlags(0)
 
-	usage := `usage: tuf [-h|--help] [-d|--dir=<dir>] <command> [<args>...]
+	usage := `usage: tuf [-h|--help] [-d|--dir=<dir>] [--insecure-plaintext] <command> [<args>...]
 
 Options:
   -h, --help
-  -d <dir>     The path to the repository (defaults to the current working directory)
+  -d <dir>              The path to the repository (defaults to the current working directory)
```
Comment (titanous) [inline tuf/main.go]: Add `, --dir`?

**Comment by titanous** [comment]
LGTM

**Pull request merged**